src/citrine/builders/descriptors.py

```diff
@@ -0,0 +1,125 @@
+from itertools import chain
+from typing import Iterator, Mapping
+
+from gemd.entity.bounds import RealBounds, CategoricalBounds, MolecularStructureBounds, \
+    IntegerBounds, CompositionBounds
+from gemd.entity.template.attribute_template import AttributeTemplate
+from gemd.entity.value import EmpiricalFormula
+
+from citrine.informatics.descriptors import RealDescriptor, CategoricalDescriptor, \
+    MolecularStructureDescriptor, Descriptor, ChemicalFormulaDescriptor
+from citrine.resources.data_concepts import DataConceptsCollection
+from citrine.resources.project import Project
+
+
+def template_to_descriptor(template: AttributeTemplate) -> Descriptor:
+    """
+    Convert a GEMD attribute template into an AI Engine Descriptor.
+
+    IntBounds cannot be converted because they have no matching descriptor type.
+    CompositionBounds cannot be converted because they may correspond to Formulations or
+    ChemicalFormulas.
```

Comment:
This doesn't match up with the current code.

Reply:
Updated